test_scripts/unit_tests/utils.bats

```diff
@@ -0,0 +1,186 @@
+setup() {
+    load '/usr/local/lib/bats-support/load.bash'
+    load '/usr/local/lib/bats-assert/load.bash'
+
+    DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd)"
+    PATH="$DIR/../../common:$PATH"
+    export NO_COLOR=1
+    source "$DIR/../../common/utils.sh"
+    source "$DIR/../../common/loggers.sh"
+}
+
+@test "generates a random string" {
+    var1=$(__generate_random_string)
+    var2=$(__generate_random_string)
+    refute [ "$var1" == "$var2" ]
+}
+
+@test "finds value in array" {
+    cities=("CHICAGO" "LONDON" "PARIS" "NEW YORK" "SAN FRANCISCO")
+    city="PARIS"
+    var=$(__elementIn "$city" "${cities[@]}" )
+    assert [ "$var" == 0 ]
+}
+
+@test "finds value not case sensitive on needle" {
+    cities=("CHICAGO" "LONDON" "PARIS" "NEW YORK" "SAN FRANCISCO")
+    city="paris"
+    var=$(__elementIn "$city" "${cities[@]}" )
+    assert [ "$var" == 0 ]
+}
+
+@test "Works with ubuntu versions" {
+    SUPPORTED_VERSIONS=("14.04" "16.04" "18.04" "20.04")
+    OS_VERSION="18.04"
+    run __elementIn "$OS_VERSION" "${SUPPORTED_VERSIONS[@]}"
+    assert_success
+    assert_output "0"
+}
+
+@test "finds value not case sensitive on haystack" {
+    cities=("CHICAGO" "LONDON" "paris" "NEW YORK" "SAN FRANCISCO")
+    city="PARIS"
+    var=$(__elementIn "$city" "${cities[@]}" )
+    assert [ "$var" == 0 ]
+}
+
+@test "does not find value in array" {
+    cities=("CHICAGO" "LONDON" "PARIS" "NEW YORK" "SAN FRANCISCO")
+    city="MIAMI"
+    var=$(__elementIn "$city" "${cities[@]}" )
+    assert [ "$var" == 1 ]
+}
+
+@test "Compare versions outputs zero for same version" {
+    output=$(__compareVersions 2.0.0 2.0.0)
+    assert [ "$output" == 0 ]
+}
+
+@test "Compare versions outputs 1 for greater version" {
+    output=$(__compareVersions 2.0.1 2.0.0)
+    assert [ "$output" == 1 ]
+}
+
+@test "Compare versions outputs -1 for lesser version" {
+    output=$(__compareVersions 2.0.0 3.0.0)
+    assert [ "$output" == "-1" ]
+}
+
+@test "Compare versions outputs 0 for improper formatted version" {
+    output=$(__compareVersions 2X0 3.0.0)
+    assert [ "$output" == "0" ]
+}
+
+@test "Compare versions outputs 0 for improper formatted comparison version" {
+    output=$(__compareVersions 2.0.0 3X0)
+    assert [ "$output" == "0" ]
+}
+
+@test "Finds closest version of same version" {
+    UBUNTU_SUPPORTED_SYNC_GATEWAY_VERSIONS=("1.5.1" "1.5.2" "2.0.0" "2.1.0" "2.1.1" "2.1.2" "2.1.3" "2.5.0" "2.5.1" "2.6.0" "2.6.1" "2.7.0" "2.7.1" "2.7.2" "2.7.3" "2.7.4" "2.8.0" "2.8.2")
```

Comment:
Could we push these to a common config to source for both this and general usage?

Reply:
That's a good idea